cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/remote/RemoteNamespaceQueryTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.store.remote;
+
+import co.cask.cdap.common.namespace.RemoteNamespaceQueryClient;
+import co.cask.cdap.gateway.handlers.meta.RemoteNamespaceQueryHandler;
+import co.cask.cdap.internal.app.services.http.AppFabricTestBase;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.NamespaceConfig;
+import co.cask.cdap.proto.NamespaceMeta;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.store.NamespaceStore;
+import com.google.inject.Injector;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+
+/**
+ * Tests implementation of {@link RemoteNamespaceQueryHandler} by using it fetch namespaces.
+ */
+public class RemoteNamespaceQueryTest extends AppFabricTestBase {
```

Comment:
do we need to extend app fabric test base? You may be able to save some test execution time by just starting the RemoteOperationsService like so - https://github.com/caskdata/cdap/pull/6090/files#diff-2e3c97b48baba50ebdeea3afcdd1b8dc. Since you're using NamespaceStore, you may need to start Transaction service and DatasetService additionally, but the others should not be needed.